test/modules/load-gen/PriorityMessageSender.cs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace LoadGen
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Extensions.Logging;
+
+    class PriorityMessageSender : SenderBase
+    {
+        readonly string[] outputs = new string[] { "pri0", "pri1", "pri2", "pri3" };
+
+        readonly Random rng = new Random();
+
+        public PriorityMessageSender(
+            ILogger logger,
+            ModuleClient moduleClient,
+            Guid batchId,
+            string trackingId)
+            : base(logger, moduleClient, batchId, trackingId)
+        {
+        }
+
+        public async override Task RunAsync(CancellationTokenSource cts, DateTime testStartAt)
+        {
+            bool firstMessageWhileOffline = true;
+            var priorityAndSequenceList = new List<(int, long)>();
```

Comment:
use dictionary<int,List<long>>

Reply:
I think SortedDictionary<int, List<long>> since the priorities need to be in ascending order for later below.  This way there's no need to re-sort later, and it just becomes list concatenation of each priority.

---
In reply to: [397497876](https://github.com/Azure/iotedge/pull/2735#discussion_r397497876) [](ancestors = 397497876)